chain/client/src/client.rs

```diff
@@ -791,7 +791,8 @@ impl Client {
         if status.is_new_head() {
             self.shards_mgr.update_largest_seen_height(block.header.inner_lite.height);
             if !self.config.archive {
-                if let Err(err) = self.chain.clear_old_data() {
+                if let Err(err) = self.chain.clear_data(self.runtime_adapter.get_trie()) {
+                    debug_assert!(false);
```

Comment:
Let's put the assert after the error message for better debugging experience should we bump into the assert.

Reply:
++